src/third_party/dartino/object_memory.cc

```diff
@@ -231,9 +231,29 @@ void Chunk::find(uword word, const char* name) {
 #endif
 
 Chunk* ObjectMemory::allocate_chunk(Space* owner, uword size) {
+#ifdef TOIT_FREERTOS
```

Comment:
I think I'd prefer the ifdef to cover the full function, so you end up with two completely separate implementations.